core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ast/constraintcomponents/QualifiedMaxCountConstraintComponent.java

```diff
@@ -83,7 +84,7 @@ public void toModel(Resource subject, IRI predicate, Model model, Set<Resource>
 		}
 
 		if (qualifiedMaxCount != null) {
-			model.add(subject, SHACL.QUALIFIED_MAX_COUNT, literal(qualifiedMaxCount));
+			model.add(subject, SHACL.QUALIFIED_MAX_COUNT, literal(qualifiedMaxCount + "", XMLSchema.INTEGER));
```

Comment:
I don't understand what's going on here. You have a `Long` value, which you then concatenate with an empty string, and you turn it into an integer? Why do this? 

Also: I'd suggest using `XSD` instead of `XMLSchema` (I think it's deprecated?)

Reply:
It's to make the output look identical to the input. This code is currently only used while running the tests, so we can change it up later if we want.

Reply:
Can you maybe add a code comment to indicate that? These kinds of things tend to stick around longer than you think, and it's not really clear just from looking at the code what it's for.

Reply:
Will do :)

Reply:
Other than that happy with this being merged btw

Reply:
Im taking a break now to go to a cafe and relax a bit. So I'll fix things up and merge this later today.

Reply:
Found out I could clean this up because of: https://github.com/eclipse/rdf4j/issues/2843

Reply:
Actually read the spec again. So I'm converting long to BigInteger on the way out. In the future we could consider using BigInteger directly.